[Havoc] Initial T31 Set Implementation

hotbees opened the pull request:
Implements the havoc 2pc and 4pc for T31

2pc still needs RNG target selection implemented for AOE lest prio damage be inflated

Split the reporting of the t31 glaives; left soulrend dot unsplit for now.

hotbees committed:
[Havoc] Initial T31 Set Implementation

* Still needs RNG target selection in AOE implemented

ToppleTheNun commented on engine/class_modules/sc_demon_hunter.cpp:
```diff
@@ -7156,6 +7192,12 @@ void demon_hunter_t::init_spells()
+
+  if ( set_bonuses.t31_havoc_2pc->ok() )
+  {
+    throw_glaive_t* throw_glaive_t31 = get_background_action<throw_glaive_t>( "throw_glaive_t31", "", true );
```
Need to make sure this action has a cooldown of `0_s`, otherwise we start getting errors.
```suggestion
    throw_glaive_t* throw_glaive_t31 = get_background_action<throw_glaive_t>( "throw_glaive_t31", "", true );
    throw_glaive_t31->internal_cooldown->base_duration = 0_s;
    throw_glaive_t31->cooldown->base_duration              = 0_s;
    throw_glaive_t31->cooldown->charges                        = 0;
```

ToppleTheNun reviewed (changes requested):
There are some assertion failures without the suggested change.

hotbees committed:
Update sc_demon_hunter.cpp

- Removed seperate reporting for now, soulrend needs to be dealt with for that.
- Added 0s charge cd to satisfy asserts
